packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -2392,14 +2392,31 @@ function pushImg(
   props: Object,
   resources: Resources,
 ): null {
+  const {src, srcSet} = props;
   if (
     props.loading !== 'lazy' &&
-    typeof props.src === 'string' &&
-    props.fetchPriority !== 'low'
+    typeof src === 'string' &&
+    props.fetchPriority !== 'low' &&
+    // We exclude data URIs in src and srcSet since these should not be preloaded
+    !(
+      src[4] === ':' &&
+      (src[0] === 'd' || src[0] === 'D') &&
+      (src[1] === 'a' || src[1] === 'A') &&
+      (src[2] === 't' || src[2] === 'T') &&
+      (src[3] === 'a' || src[3] === 'A')
```

Comment:
```suggestion
      src[0].toLowerCase() === 'd' &&
      src[1].toLowerCase() === 'a' &&
      src[2].toLowerCase() === 't' &&
      src[3].toLowerCase() === 'a'
```

Reply:
I think you'll find this is slower if you test it